core/src/main/java/com/novoda/noplayer/internal/Heart.java

```diff
@@ -78,7 +78,9 @@ public Heartbeat(NoPlayer.HeartbeatCallback callback, NoPlayer player) {
 
         @Override
         public void run() {
-            callback.onBeat(player);
+            if (player.isPlaying()) {
```

Comment:
Why not starting/stopping the heartbeat when the playback starts/end instead? Seems more straightforward

Reply:
We do that already. The problem is that this handler might execute between player state changes 😢 it's kind of an edge case but we have seen it.

Reply:
do you think we need to log/call a listener to let clients know this heartbeat has been skipped?

Reply:
I'm not sure tbh, I think emitting when it isn't playing is an error, it shouldn't happen. We would just be telling clients about behaviour we would never expect them to deal with. What do you think?

Reply:
I agree, just a little paranoid that we might have changed behaviour (by fixing bad state).

I'd be happy with release notes that explain the scenario